contracts/contracts/strategies/AaveStrategy.sol

```diff
@@ -170,19 +213,71 @@ contract AaveStrategy is InitializableAbstractStrategy {
     }
 
     /**
-     * @dev Get the current address of the Aave lending pool core, which stores all the
-     *      reserve tokens in its vault.
-     * @return Current lending pool core address
+     * @dev Collect stkAave, convert it to AAVE send to Vault.
      */
-    function _getLendingPoolCore() internal view returns (address payable) {
-        address payable lendingPoolCore = ILendingPoolAddressesProvider(
-            platformAddress
-        )
-            .getLendingPoolCore();
-        require(
-            lendingPoolCore != address(uint160(address(0))),
-            "Lending pool core does not exist"
-        );
-        return lendingPoolCore;
+    function collectRewardToken() external onlyVault nonReentrant {
+        if (address(stkAave) == address(0)) {
+            return;
+        }
+
+        // Check staked AAVE cooldown timer
+        uint256 cooldown = stkAave.stakersCooldowns(address(this));
+        uint256 windowStart = cooldown + stkAave.COOLDOWN_SECONDS();
+        uint256 windowEnd = windowStart + stkAave.UNSTAKE_WINDOW();
+
+        // If inside the unlock window, then we can redeem stkAave
+        // for AAVE and send it to the vault.
+        if (now > windowStart && now <= windowEnd) {
+            // Redeem to AAVE
+            uint256 stkAaveBalance = stkAave.balanceOf(address(this));
+            if (stkAaveBalance > rewardLiquidationThreshold) {
+                stkAave.redeem(address(this), stkAaveBalance);
+            }
+            // Transfer AAVE to vaultAddress
+            uint256 aaveBalance = IERC20(rewardTokenAddress).balanceOf(
+                address(this)
+            );
+            if (aaveBalance > 0) {
+                IERC20(rewardTokenAddress).safeTransfer(
```

Comment:
The Uniswap path that we use to liquidate these is hardcoded as reward token -> weth -> usdt. Did you check if that is suitable here (and for Compound for that matter). My guess is that it is, but I've always had a bit of an uneasy feeling about it.

Reply:
I checked and that's currently the best route on both AAVE and COMP. Looks like were are good.